lib/LedgerSMB/Upgrade_Tests.pm

```diff
@@ -546,6 +619,104 @@ number of "AR_*", "AP_*" and/or "IC_*" links concatenated by colons (:).'),
     max_version => '3.0'
     );
 
+push @tests,__PACKAGE__->new(
+    test_query => q{select *, (select array_agg(x in (
+                            SELECT value FROM pg_temp.account_link
+                           )) from unnest(string_to_array(c.link,':')) x)
+                        from chart c
+                       where ((select array_agg(x in (
+                            SELECT value FROM pg_temp.account_link
+                           )) from unnest(string_to_array(c.link,':')) x) @> ARRAY[0]::bool[]
+                           or c.link is null
+                           or c.link = '')
+                         and charttype = 'A'
+                    order by accno;
+                    },
+    display_name => marktext('No unlinked account'),
+    name => 'no_unlinked_account',
+    display_cols => ['accno', 'category', 'description', 'link'],
+ instructions => marktext(
+                   'Please make sure that all accounts are linked.<br>Multiple links can be selected'),
+   columns => ['link'],
+selectable_values => {
+                        link => {
+                              query => q{SELECT * FROM pg_temp.account_link
+                                         ORDER BY text},
+                        multiselect => ':'
+                      },
+                      category => q{SELECT * FROM pg_temp.account_category
+                                    ORDER BY text}
+                     },
+    table => 'chart',
+    appname => 'sql-ledger',
+    min_version => '2.7',
+    max_version => '3.0'
+    );
+
+push @tests,__PACKAGE__->new(
+    test_query => q{select *
+                    from chart
+                   where charttype = 'A'
+                     and  category = 'A'
+                     and exists (
+                         select 1 from ap having count(*) > 0
+                     )
+                     and not exists (
+                         select 1 from chart
+                         group by link
+                         having link = 'AR'
+                     )
+                     order by accno
+                     },
+    display_name => marktext('No AR account'),
+    name => 'no_ar_account',
+    display_cols => ['accno', 'description', 'link'],
+ instructions => marktext(
+                   'Please make sure one AR account exists'),
```

Comment:
Maybe we need to eliminate the question about the default AR account instead; asking this question *only* when there are multiple AR accounts?
My point: if I don't need an AR account for my use of LedgerSMB, then ... Why would the migration force me to create one?

Reply:
A very good idea if there is an easy way to check when they are required. SL Customers and vendors are mapped in LSMB entities, so we would been database specific checks.
Currently, we were advocating the need for a default AR and AP acount, so this code only enforces that statement and default automatically when only 1 is present in a category.

Reply:
We advocate having one, but what happens if there isn't one? Or when I don't select one? Do things fail?